### ISSUE TITLE
No "Op code completion with documentation"

### ISSUE THREAD

**Issue by evx73** [BlairLeduc/vscode-6x09-assembly#6]
Code completion and/or documentation not working. I cannot see any obvious way to enable it. Thanks for the add-on!

My setup:

Version: 1.60.0 (user setup)
Commit: e7d7e9a9348e6a8cc8c03f877d39cb72e5dfb1ff
Date: 2021-09-01T10:41:52.311Z
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19043

**Comment by evx73** [comment]
The "Log (Extension Host)" is full of similar log lines;

```
[2021-09-11 23:34:02.096] [exthost] [error] [blairleduc.6x09-assembly] provider FAILED
[2021-09-11 23:34:02.096] [exthost] [error] TypeError: Cannot read property 'lines' of undefined
	at c:\Users\evx\.vscode\extensions\blairleduc.6x09-assembly-0.6.7\out\providers\hover.js:29:59
	at new Promise (<anonymous>)
	at HoverProvider.provideHover (c:\Users\evx\.vscode\extensions\blairleduc.6x09-assembly-0.6.7\out\providers\hover.js:23:16)
	at g.provideHover (c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:94:166338)
	at c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:94:190870
	at xe._withAdapter (c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:94:187656)
	at xe.$provideHover (c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:94:190848)
	at i._doInvokeHandler (c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:98:13543)
	at i._invokeHandler (c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:98:13227)
	at i._receiveRequest (c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:98:11838)
	at i._receiveOneMessage (c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:98:10567)
	at c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:98:8663
	at c.fire (c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:66:1712)
	at l.fire (c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:74:15781)
	at c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:30187
	at c.fire (c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:66:1712)
	at l.fire (c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:74:15781)
	at r._receiveMessage (c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:74:21051)
	at c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:74:17929
	at c.fire (c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:66:1712)
	at v.acceptChunk (c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:74:13142)
	at c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:74:12490
	at Socket.w (c:\Users\evx\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:113:13350)
	at Socket.emit (events.js:315:20)
	at addChunk (internal/streams/readable.js:309:12)
	at readableAddChunk (internal/streams/readable.js:284:9)
	at Socket.Readable.push (internal/streams/readable.js:223:10)
	at Pipe.onStreamRead (internal/stream_base_commons.js:188:23)
```

**Comment by evx73** [comment]
I quickly debugged the extension. It seems that `vscode.workspace.getWorkspaceFolder` fails to return a folder unless the file uri was opened from the workspace. For example, it does not work if the file is open via the `File > Open File...` menu.

It fails in `out\managers\workspace.js`

```
getOrCreateFolder(document) {
        const workspaceFolder = vscode.workspace.getWorkspaceFolder(document.uri);
        ...
```

Possibly related to this? https://github.com/microsoft/vscode/issues/36221

**Comment by BlairLeduc** [comment]
Apologies for the late reply. Something is definitely rotten in workspaces and opening single files. I need to look at this but I have been overrun at my day job. I will be getting back to this in the coming months.

**Comment by BlairLeduc** [comment]
Apologies for the late response. Something is definitely rotten here. I have been overrun at my day job and moving to the UK. I hope to get back to this in the coming months.

**Comment by BlairLeduc** [comment]
This will be fixed in 0.7.0

**Comment by evx73** [comment]
Super, thank you!
